samples/basic/blinky_pwm/sample.yaml

```diff
@@ -8,3 +8,4 @@ tests:
       - pwm
     depends_on: pwm
     harness: led
+    platform_exclude: rpi_pico/rp2040/w
```

Comment:
why doesn;t the filter above work? this seems like a hack

Reply:
I can accept this as a current solution, as the docs for the W variant states:
> Since GPIO 25 is routed to the on-board LED on the Pico, but to the Infineon module on the Pico W, the “blinky” sample program does not work on the W

Ref: https://docs.zephyrproject.org/latest/boards/arm/rpi_pico/doc/index.html

But we should consider the cases where a base dts for a board variant removes nodes which may be adjusted bases on higher level board overlays, like in this case.

Current behavior for picking up overlays is to facilitate moving from hwm v1 to v2, as well as supporting common overlays to minimize need for n-overlays when a single can do.

Reply:
I also agree @gmarull points.

https://github.com/zephyrproject-rtos/zephyr/pull/69025#pullrequestreview-1882726300

It needs to be removed on the w case, but using a filter is preferable.